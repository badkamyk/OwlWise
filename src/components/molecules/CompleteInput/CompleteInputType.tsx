import {InputProps} from "../../atoms/Input/InputType";
import {InputLabelType} from "../../atoms/InputLabel/InputLabelType";

export type CompleteInputType = InputProps & Omit<InputLabelType, "htmlFor">;