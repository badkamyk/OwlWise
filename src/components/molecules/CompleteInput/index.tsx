import InputLabel from "../../atoms/InputLabel";
import Input from "../../atoms/Input";
import { CompleteInputType } from "./CompleteInputType";

export default function CompleteInput({ text, type, id, placeholder, required }: CompleteInputType) {
	return (
		<div>
			<InputLabel text={text} htmlFor={id} />
			<Input {...{ type, id, placeholder, required }} />
		</div>

	);
}