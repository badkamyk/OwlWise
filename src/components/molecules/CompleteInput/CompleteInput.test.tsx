import { render, renderHook, screen } from "@testing-library/react";
import { FormProvider, useForm } from "react-hook-form";
import { expect, it } from "vitest";
import CompleteInput from "./index";

it("CompleteInput", () => {
	const { result } = renderHook(() => useForm());
	const { container } = render(
		<FormProvider {...result.current}>
			<CompleteInput
				text="Email"
				type="email"
				id="email"
				placeholder="Your email"
				required
			/>
		</FormProvider>
	);
	expect(screen.getByText("Email")).toBeTruthy();
	expect(screen.getByLabelText("Email")).toBeTruthy();
	expect(screen.getByRole("textbox")).toHaveProperty("id", "email");
	expect(screen.getByRole("textbox")).toHaveProperty(
		"placeholder",
		"Your email"
	);
	expect(screen.getByRole("textbox")).toHaveProperty("required", true);
	expect(container).toMatchSnapshot();
});
