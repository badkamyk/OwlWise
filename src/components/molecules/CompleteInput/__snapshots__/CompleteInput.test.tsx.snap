// Vitest Snapshot v1

exports[`CompleteInput 1`] = `
<div>
  <div>
    <label
      class="mb-2 block text-sm font-medium text-gray-900 dark:text-gray-300"
      for="email"
    >
      Email
    </label>
    <input
      aria-invalid="false"
      class="focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-500 dark:focus:border-primary-500 dark:shadow-sm-light block w-full rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-900 shadow-sm dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400"
      id="email"
      name="email"
      placeholder="Your email"
      required=""
      type="email"
    />
  </div>
</div>
`;
